src/build_packages.sh

```diff
@@ -2,27 +2,58 @@
 set -e
 export XROAD=$(cd "$(dirname "$0")"; pwd)
 
-if [[ ! " $* " =~ " --packageonly " ]]; then
-  ./compile_code.sh "$@"
-fi
+HAS_DOCKER=""
+
+errorExit() {
+	echo "*** $*" 1>&2
+	exit 1
+}
+
+buildInDocker() {
+    test -n "$HAS_DOCKER" || errorExit "Error, docker is not installed/running."
+    echo "Building in docker..."
+    BUILD_IMAGE="`docker images | grep xroad-build | awk '{ print $1 }'`"
+    if [ "$BUILD_IMAGE" != "xroad-build" ]; then
+        echo "Build image 'xroad-build' not found, building it."
+        docker build -q -t xroad-build --build-arg uid=$(id -u) --build-arg gid=$(id -g) $XROAD/packages/docker-compile || errorExit "Error building build image."
+    fi
```

Comment:
Is this "check if image is already built" logic actually needed?

The regular docker image builds (such as ones used for packaging, https://github.com/nordic-institute/X-Road/blob/develop/src/build_packages.sh#L10-L13) already skip building if there are no changes and use build cache https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#leverage-build-cache instead.

Using the regular build also has the advantage that it notices if image **does** need to be rebuild (e.g. changes to Dockerfile), which this does not.

Reply:
That's fine for me.